### ISSUE TITLE
[tank game] RustError("A script in this movie has taken too long to execute and has been terminated.")

### ISSUE THREAD

**Issue by olux997** [ruffle-rs/ruffle#12344]
### Describe the bug

ruffle_web-wasm_extensions.js:719 ERROR core/src/avm2/events.rs:419 Error dispatching event EventObject(EventObject { type: "complete", class: flash.events::Event, ptr: 0x13bb9c8 }) to handler FunctionObject(FunctionObject { ptr: 0x10a03e0 }) : RustError("A script in this movie has taken too long to execute and has been terminated.")

Game's SWF
[tank.zip](https://github.com/ruffle-rs/ruffle/files/12179707/tank.zip)

### Expected behavior

ruffle_web-wasm_extensions.js:719 ERROR core/src/avm2/events.rs:419 Error dispatching event EventObject(EventObject { type: "complete", class: flash.events::Event, ptr: 0x13bb9c8 }) to handler FunctionObject(FunctionObject { ptr: 0x10a03e0 }) : RustError("A script in this movie has taken too long to execute and has been terminated.")

### Affected platform

Self-hosted version

### Operating system

Windows 10

### Browser

Google Chrome 114.0.5735.199 [64 bits]

**Comment by n0samu** [comment]
It seems like the game is going into an infinite loop and using more and more memory. On the desktop app, it eventually fills up all of my RAM without visibly progressing.

**Comment by olux997** [comment]
> It seems like the game is going into an infinite loop and using more and more memory. On the desktop app, it eventually fills up all of my RAM without visibly progressing.
This game just can run at [http://] , can't run at [desktop app]....
demo: [http://www.xxx.com/tank.swf] ....flashplayer can't run direct...

**Comment by n0samu** [comment]
Yes, I used Ruffle's `--dummy-external-interface` option to bypass that requirement.

**Comment by olux997** [comment]
> 

:
01. why https://github.com/ruffle-rs/ruffle/wiki/Using-Ruffle#configuration-options  [ no --dummy-external-interface option ]
02. Can work about SWF if use --dummy-external-interface ?

**Comment by n0samu** [comment]
The Ruffle wiki page only lists web options, not command-line options for the desktop app. I was explaining how I got the game to work in the desktop app. To see all of the command line options, run `ruffle --help`. Also, the options can be set from the "Open Advanced" menu now as well.

**Comment by olux997** [comment]
> The Ruffle wiki page only lists web options, not command-line options for the desktop app. I was explaining how I got the game to work in the desktop app. To see all of the command line options, run `ruffle --help`. Also, the options can be set from the "Open Advanced" menu now as well.

:
01. I found that the DESKTOP APP version runs much faster than the WEB version
02. How can I run this game on the Self_host version?

**Comment by n0samu** [comment]
I shouldn't have said "get the game to work", I just meant "reproduce the bug". The game doesn't work in Ruffle, either on desktop or web.
